test/integration_test.exs

```diff
@@ -0,0 +1,251 @@
+              index: 0,
+              id: "stream-icons",
+              envelope: %{
+                head: [],
+                bodyLast: [],
+                bodyInline: "<DIV id=\"site-container\">"
+              }
+            }
+          ]
+        })
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "when fetching a big component" do
+    test "it will return a JSON response" do
+      json_body = ~s({
+        "components": [{
+                        "id": "big-component",
+                        "endpoint": "localhost:8082/big_component",
+                        "must_succeed": true
+                        }]
+                      }
+                    )
+      conn = conn(:post, "/collect", json_body)
+      conn = Router.call(conn, @opts)
+
+      expected_body =
+        Jason.encode!(%{
+          components: [
+            %{
+              status: 200,
+              index: 0,
+              id: "big-component",
+              envelope: %{
+                head: [],
+                bodyLast: [],
+                bodyInline: "<DIV id=\"big-component\" class=\"big\"><h1>This is a really big component</h1></DIV>"
+              }
+            }
+          ]
+        })
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
+      assert conn.resp_body == expected_body
+    end
+  end
+
+  describe "fetching multiple components" do
+    test "when all of the components are successful they all have a status of 200" do
+      json_body = ~s({
+        "components": [{ "id": "news-front-page",
+                        "endpoint": "localhost:8082/success",
+                        "must_succeed": true
+                        },
+                        {
+                          "id": "weather-front-page",
+                          "endpoint": "localhost:8082/success",
+                          "must_succeed": true
+                        },
+                        {
+                          "id": "weather-component",
+                          "endpoint": "localhost:8082/big_component",
+                          "must_succeed": true
+                        }]
+                      }
+                    )
+      conn = conn(:post, "/collect", json_body)
+      conn = Router.call(conn, @opts)
+
+      expected_body =
+        Jason.encode!(%{
+          components: [
+            %{
+              status: 200,
+              index: 0,
+              id: "news-front-page",
+              envelope: %{
+                head: [],
+                bodyLast: [],
+                bodyInline: "<DIV id=\"site-container\">"
+              }
+            },
+            %{
+              status: 200,
+              index: 0,
```

Comment:
Don't we expect this to be index 1?

Reply:
There is a ticket for this. 

https://jira.dev.bbc.co.uk/browse/RESFRAME-2699